src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -962,6 +965,79 @@ func (r *ClusterReconciler) setInitialSuperUserPassword(
 	return utilerrors.NewAggregate(errs)
 }
 
+// decommissionGhostBrokers decommissions brokers that redpanda thinks exists, but aren't assigned to any pods
+// This is not a reversible process. If creating a new broker due to an empty disk was a mistake, the data
+// that was on that disk will be unusable.
+func (r *ClusterReconciler) decommissionGhostBrokers(c context.Context, vCluster *vectorizedv1alpha1.Cluster, l logr.Logger, ar *attachedResources) error {
+	if r.GhostDecommissioning {
+		return r.doDecommissionGhostBrokers(c, vCluster, l, ar)
+	}
+	return nil
+}
+
+// custom error to satisfy err113
+type missingBrokerID struct{}
+
+func (m *missingBrokerID) Error() string {
+	return "a pod is temporarily missing the broker-id annotation"
+}
+
+func (r *ClusterReconciler) doDecommissionGhostBrokers(c context.Context, vCluster *vectorizedv1alpha1.Cluster, l logr.Logger, ar *attachedResources) error {
+	ctx, done := context.WithCancel(c)
+	defer done()
+	log := l.WithName("deleteGhostBrokers")
+	log.V(logger.DebugLevel).Info("deleting ghost brokers")
+
+	pods, err := r.podList(ctx, vCluster)
+	if err != nil {
+		return fmt.Errorf("unable to fetch PodList: %w", err)
+	}
+
+	pki, err := ar.getPKI()
+	if err != nil {
+		return fmt.Errorf("getting pki: %w", err)
+	}
+
+	adminClient, err := r.AdminAPIClientFactory(ctx, r.Client, vCluster, ar.getHeadlessServiceFQDN(), pki.AdminAPIConfigProvider())
+	if err != nil {
+		return fmt.Errorf("unable to create admin client: %w", err)
+	}
+	adminBrokers, err := adminClient.Brokers(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to fetch brokers: %w", err)
+	}
+
+	actualBrokerIDs := make(map[string]any, len(pods.Items))
+	for i := range pods.Items {
+		pod := &pods.Items[i]
+		if pod.Annotations == nil {
+			pod.Annotations = make(map[string]string)
+		}
+		nodeIDStrAnnotation, annotationExist := pod.Annotations[resources.PodAnnotationNodeIDKey]
+		if !annotationExist {
+			return fmt.Errorf("requeuing: %w", &missingBrokerID{})
```

Comment:
These are always a toss up for me, do we wish to collect errors as we go and have this be a continue, and try to process everything now, or is better to stop now and requeue?

Reply:
By the time we get here, all the pods should have their id annotation. If they don't, something unexpected happened and we're not in the state we think we are, so we should bail and start the reconcile over.